RELEASE/scripts/autoscend/auto_providers.ash

```diff
@@ -1212,3 +1212,252 @@ boolean provideMoxie(int amt, boolean doEquips)
+{
+	auto_log_info((speculative ? "Checking if we can" : "Trying to") + " provide " + amt + " meat, " + (doEverything ? "with" : "without") + " equipment, familiar, and limited buffs", "blue");
+	float alreadyHave = numeric_modifier("Meat Drop");
+	float need = amt - alreadyHave;
+	if(need > 0)
+	{
+		auto_log_debug("We currently have " + alreadyHave + ", so we need an extra " + need);
+	}
+	else
+	{
+		auto_log_debug("We already have enough +meat!");
+		return alreadyHave;
+	}
+	float delta = 0;
+	float result()
+	{
+		return numeric_modifier("Meat Drop") + delta;
+	}
+	boolean pass()
+	{
+		return result() >= amt;
+	}
+	if(pass())
+		return result();
+	
+	// don't craft equipment here. See how much +item we can get with gear on hand
+	if(doEverything)
+	{
+		string max = "500meat " + (amt + 100) + "max";
+		if(speculative)
+		{
+			simMaximizeWith(loc, max);
+		}
+		else
+		{
+			addToMaximize(max);
+			simMaximize(loc);
+		}
+		delta = simValue("Meat Drop") - numeric_modifier("Meat Drop");
+		auto_log_debug("With existing gear we can get to " + result());
+		if(pass())
+			return result();
+	}
+	//see how much familiar will help
+	if(doEverything && pathHasFamiliar() && pathAllowsChangingFamiliar())
+	{
+		if(!speculative)
+		{
+			handleFamiliar("meat");
+		}
+		// fam isn't equipped immediatly even if we aren't speculating
+		// so add bonus from fam regardless of speculation
+		familiar target = lookupFamiliarDatafile("meat");
+		if(target != $familiar[none] && target != my_familiar())
+		{
+			int famWeight = familiar_weight(target) + weight_adjustment();
+			delta += numeric_modifier(target, "Meat Drop",famWeight,$item[none]);
+			auto_log_debug("With using familiar: " + target + " we can get to " + result());
+		}
+		else
+		{
+			auto_log_debug("Already have desired familar, " + target + ", active.");
+		}
+		if(pass())
+			return result();
+	}
+	void handleEffect(effect eff)
+	{
+		if(speculative)
+		{
+			delta += numeric_modifier(eff, "Meat Drop");
+		}
+		auto_log_debug("We " + (speculative ? "can gain" : "just gained") + " " + eff.to_string() + ", now we have " + result());
+	}
+	boolean tryEffects(boolean [effect] effects)
+	{
+		foreach eff in effects
+		{
+			if(buffMaintain(eff, 0, 1, 1, speculative))
+				handleEffect(eff);
+			if(pass())
+				return true;
+		}
+		return false;
+	}
+	if((get_property("sidequestArenaCompleted") == "fratboy") && !get_property("concertVisited").to_boolean() && (have_effect($effect[Winklered]) == 0))
+	{
+		if(is_professor())
+		{
+			//Need to manually equip because professor
+			if(!have_equipped($item[beer helmet])) equip($item[beer helmet]);
+			if(!have_equipped($item[distressed denim pants])) equip($item[distressed denim pants]);
+			if(!have_equipped($item[bejeweled pledge pin])) equip($item[bejeweled pledge pin]);
+		}
+		else
+		{
+			outfit("Frat Warrior Fatigues");
+		}
+		cli_execute("concert 2");
+	}
+	handleBjornify($familiar[Hobo Monkey]);

```

Comment:
I'm not familiar with bjorn. Do we need to calculate if putting hobo monkey in bjorn is better than simply having hobo the active fam? Since this uses an equip slot, seems like this should only happen if `doEverything` is true. I'd also consider moving it down farther as well since it takes an equipment slot